AsyncDisplayKit/Details/ASCollectionDataController.mm

```diff
@@ -271,6 +319,14 @@ - (ASCellNode *)supplementaryNodeOfKind:(NSString *)kind atIndexPath:(NSIndexPat
   return nil;
 }
 
+- (id<ASSectionContext>)contextForSectionAtIndex:(NSInteger)sectionIndex
+{
+  ASDisplayNodeAssertMainThread();
+  ASDisplayNodeAssertTrue(_dataSourceSupportsSectionContext);
+  ASDisplayNodeAssertTrue(sectionIndex >= 0 && sectionIndex < _sectionContexts.count);
+  return _sectionContexts[sectionIndex];
```

Comment:
I think we can be lenient and not assert that the data source supports section contexts here.